doc/md/versioned/troubleshoot.mdx

```diff
@@ -0,0 +1,289 @@
+    (1, 2), (2, 2), (3, 2),
+    -- Constraint error (adding masseelch twice to seniors)
+    (1, 2);
+// highlight-next-line-error-message
+     Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+
+  -------------------------
+  -- 2.521007ms
+  -- 0 migrations ok (1 with errors)
+  -- 2 sql statements ok (1 with errors)
+
+// highlight-next-line-error-message
+Error: Execution had errors:
+// highlight-next-line-error-message
+Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+<TabItem value="apply_stderr">
+
+```text
+$ atlas --env local migrate apply 1 --tx-mode none
+Error: sql/migrate: execute: executing statement "INSERT INTO `user_groups` (`user_id`, `group_id`) VALUES
+    -- Founders\n    (2, 1), (3, 1),
+    -- Seniors\n    (1, 2), (2, 2), (3, 2),
+    -- Constraint error (adding masseelch twice to seniors)
+    (1, 2);" from version "2": Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+<TabItem value="status">
+
+```text
+$ atlas --env local migrate status
+Migration Status: PENDING
+  -- Current Version: 2 (2 statements applied)
+  -- Next Version:    2 (1 statements left)
+  -- Executed Files:  2 (last one partially)
+  -- Pending Files:   2
+
+Last migration attempt had errors:
+  -- SQL:   INSERT INTO `user_groups` (`user_id`, `group_id`) VALUES     -- Founders     (2, 1), (3, 1),     -- Seniors     (1, 2), (2, 2), (3, 2),     -- Constraint error (adding masseelch twice to seniors)     (1, 2);
+// highlight-next-line-error-message
+  -- ERROR: Error 1062: Duplicate entry '1-2' for key 'user_groups.PRIMARY'
+```
+
+</TabItem>
+</Tabs>
+
+## Fixing Migration Failures
+
+In our experience, the most common failures are introduced by one of the following issues:
+
+- **Syntax Error**: One SQL statement contained a syntax error. Since Atlas provides a way to [lint](/versioned/lint) a
+migration directory, this case happens due to a misconfiguration, e.g. not executing the linter before applying the
+migration or applying against the wrong dialect (like SQLite files against MySQL).
+- **Schema Dependant Change**: One SQL statement was incorrect in schema context, e.g. adding a table that does already
+exist. This case is usually caused by a schema drift, e.g. because someone manually manipulated the database schema.
+- **Data Dependant Change**: One SQL statement contained a data dependant change. If your migration adds a constraint
+like `NOT NULL` and the column does contain `NULL` values, the migration will throw an error. Though Atlas' linting
+will warn if such a change is detected, this is still a possible scenario that could break migration execution.
+- **Connection Loss**: For a number of reasons the connection to the database could be interrupted, e.g. due to a
+network failure or database crash.
+
+### Connection Loss Failures
+
+Just re-try migration execution. Atlas is smart enough to figure out where to proceed execution. For very rare cases,
```

Comment:
"smart enough" feels foreign in a document with this level of detail. 
Atlas stores metadata about executed statements and can resume without re-running statements that succeeded, or something like this